invenio_records_resources/errors.py

```diff
@@ -10,10 +10,40 @@
 
 import json
 
+from flask import abort, current_app, jsonify, make_response, request, url_for
 from invenio_rest.errors import RESTException, RESTValidationError
+from werkzeug.routing import BuildError
 
 # TODO: Revise this
 
+
+def create_pidredirected_error_handler(route):
+    """Creates an error handler for `PIDRedirectedError` error."""
+
+    def pid_redirected_error_handler(e):
+        try:
+            # Check that the source pid and the destination pid are of the same
+            # pid_type
+            assert e.pid.pid_type == e.destination_pid.pid_type
+            # Redirection works only for the item route of the format
+            # `/records/<pid_value>`
+            location = url_for(
```

Comment:
shelvable: at business layer we are trying to build links differently (to avoid ties to views), maybe we should open an issue to have this consistent accross the codebase.

Reply:
I went for the request as the view name is not retrievable from the `ResourceConfig`. Still, using the `request` is fine as we are handling errors that we want to return a response back. Which is your take on how to ideally build this?

Reply:
I dont have one, that's why `shelve`. My point was more on "this is yet another case of inconsistency" should we list them in an issue concerning "link building"?

Reply:
Ok I will create a general one regarding link building :)

Reply:
@ppanero https://github.com/inveniosoftware/invenio-records-resources/issues/45. You can add more code cases there :)